pkg/ingester/activeseries/active_series.go

```diff
@@ -41,13 +41,14 @@ type seriesStripe struct {
 
 	mu             sync.RWMutex
 	refs           map[uint64][]seriesEntry
-	active         int   // Number of active entries in this stripe. Only decreased during purge or clear.
-	activeMatching []int // Number of active entries in this stripe matching each matcher of the configured Matchers.
+	active         map[string]int // Number of active entries in this stripe per group. Only decreased during purge or clear.
```

Comment:
I assume there's some performance impact of moving from the plain integer to a map. Because the vast majority of tenants will probably not be using this feature and the active series tracker is provisioned per-tenant, we could keep the old integer variable and only use the map if the tenant is using the separating metrics feature. Not sure if that's worth it thought.